src/App.jsx

```diff
@@ -1,23 +1,26 @@
-/* eslint-disable no-undef */
 /* eslint-disable no-unused-vars */
-// eslint-disable-next-line no-unused-vars
-import React, { useState } from 'react';
+
+import React, {useState, useEffect} from 'react';
 import TodoList from './TodoList';
 import AddTodoForm from './AddTodoForm';
 
+const useSemiPState = (key, initialState) => {
+  const [state, setState] = React.useState(() => {
+    const savedValue = localStorage.getItem(key); 
+    return savedValue ? JSON.parse(savedValue) : initialState; 
+  })
+
+  React.useEffect(() => {
+    localStorage.setItem(key,JSON.stringify(state)); 
+  }, [key, state]); 
+
+    return [state, setState]; 
```

Comment:
I think your issue might be with the dependency on `[key, state]`. I think the dependency on `key` could be causing excessive calls to the `localStorage.setItem` . I would remove the `key, intialState` parameters from you function and directly pass in `savedTodoList` as the string key value to the `setItem` and `getItem` fn calls.